pkg/util/duration/duration.go

```diff
@@ -617,16 +617,35 @@ func (d Duration) MulFloat(x float64) Duration {
 
 // DivFloat returns a Duration representing a time length of d/x.
 func (d Duration) DivFloat(x float64) Duration {
-	monthInt, monthFrac := math.Modf(float64(d.Months) / x)
-	dayInt, dayFrac := math.Modf((float64(d.Days) / x) + (monthFrac * DaysPerMonth))
+	// In order to keep it compatible with PostgreSQL, we use the same logic.
+	// Refer to https://github.com/postgres/postgres/blob/e56bce5d43789cce95d099554ae9593ada92b3b7/src/backend/utils/adt/timestamp.c#L3266-L3304.
+	month := int32(float64(d.Months) / x)
+	day := int32(float64(d.Days) / x)
+
+	remainderDays := (float64(d.Months)/x - float64(month)) * DaysPerMonth
+	remainderDays = secRoundToEven(remainderDays)
+	secRemainder := (float64(d.Days)/x - float64(day) +
+		remainderDays - float64(int64(remainderDays))) * SecsPerDay
+	secRemainder = secRoundToEven(secRemainder)
+	if math.Abs(secRemainder) >= SecsPerDay {
+		day += int32(secRemainder / SecsPerDay)
+		secRemainder -= float64(int32(secRemainder/SecsPerDay) * SecsPerDay)
+	}
+	day += int32(remainderDays)
+	microSecs := float64(time.Duration(d.nanos).Microseconds())/x + secRemainder*MicrosPerMilli*MillisPerSec
+	ret := time.Duration(int64(math.RoundToEven(microSecs))) * time.Microsecond
```

Comment:
nit: i don't think we should call this `ret`. it should be `secsDuration`, or something more descriptive. we are not returning `ret`